litebird_sim/mpi.py

```diff
@@ -22,10 +22,47 @@ class _SerialMpiCommunicator:
     size = 1
 
 
+class _GridCommClass:
+    """
+    This class encapsulates the `COMM_OBS_GRID` and `COMM_NULL` communicators. It
+    offers explicitly defined setter functions so that the communicators cannot be
+    changed accidentally.
+
+    Attributes:
+
+        COMM_OBS_GRID (mpi4py.MPI.Intracomm): A subset of `MPI.COMM_WORLD` that
+            contain all the processes associated with non-zero observations.
+
+        COMM_NULL (mpi4py.MPI.Comm): A NULL communicator. When MPI is not enabled, it
+            is set as `None`. If MPI is enabled, it is set as `MPI.COMM_NULL`
+
+    """
+
+    def __init__(self, comm_obs_grid=_SerialMpiCommunicator(), comm_null=None):
+        self._MPI_COMM_OBS_GRID = comm_obs_grid
+        self._MPI_COMM_NULL = comm_null
+
+    @property
+    def COMM_OBS_GRID(self):
+        return self._MPI_COMM_OBS_GRID
+
+    @property
+    def COMM_NULL(self):
+        return self._MPI_COMM_NULL
+
+    def _set_comm_obs_grid(self, comm_obs_grid):
+        self._MPI_COMM_OBS_GRID = comm_obs_grid
+
+    def _set_null_comm(self, comm_null):
+        self._MPI_COMM_NULL = comm_null
+
+
 #: Global variable equal either to `mpi4py.MPI.COMM_WORLD` or a object
 #: that defines the member variables `rank = 0` and `size = 1`.
 MPI_COMM_WORLD = _SerialMpiCommunicator()
 
+comm_grid = _GridCommClass()
+
```

Comment:
Being a global variable that is not expected to be reassigned, I would name it `MPI_COMM_GRID`, in analogy with `MPI_COMM_WORLD`, `MPI_ENABLED`, etc.

Reply:
Refactoring a variable name is trivial if you use an advanced IDE like PyCharm. If you don't, let me know, and I'll rename it. (It’s just a matter of one mouse click.)

Reply:
> Being a global variable that is not expected to be reassigned, I would name it `MPI_COMM_GRID`, in analogy with `MPI_COMM_WORLD`, `MPI_ENABLED`, etc.

Do you mean removing the `_GridCommClass` altogether and instead simply use `MPI_COMM_GRID`?

Reply:
> > Being a global variable that is not expected to be reassigned, I would name it `MPI_COMM_GRID`, in analogy with `MPI_COMM_WORLD`, `MPI_ENABLED`, etc.
> 
> Do you mean removing the `_GridCommClass` altogether and instead simply use `MPI_COMM_GRID`?

No, it's just a matter of renaming `comm_grid`, as in this way it will match the naming convention of the other global variables in the `mpi` module.